src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -101,6 +101,9 @@ internal NpgsqlTypeHandler GetByClrType(Type type)
             if (_byClrType.TryGetValue(type, out var handler))
                 return handler;
 
+            if (Nullable.GetUnderlyingType(type) is Type underlyingType && _byClrType.TryGetValue(underlyingType, out handler))
```

Comment:
Unless I'm mistaken, this means that it was previously impossible to write a nullable value without specifying NpgsqlDbType/DbType, right? If so, can we add a test for that?

Reply:
Yes, there were no way to detect the type based on a `null` value. The generic parameter was unable to handle nulls too. Will add tests tomorrow.